core/ledger/kvledger/snapshot.go

```diff
@@ -0,0 +1,199 @@
+		return errors.Wrapf(err, "error while renaming dir [%s] to [%s]:", snapshotTempDir, slgrht)
+	}
+	if err := syncParentDir(slgrht); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (l *kvLedger) generateSnapshotMetadataFiles(
+	dir string,
+	txIDsExportSummary,
+	configsHistoryExportSummary,
+	stateDBExportSummary map[string][]byte) error {
+	// generate metadata file
+	filesAndHashes := map[string]string{}
+	for fileName, hashsum := range txIDsExportSummary {
+		filesAndHashes[fileName] = hex.EncodeToString(hashsum)
+	}
+	for fileName, hashsum := range configsHistoryExportSummary {
+		filesAndHashes[fileName] = hex.EncodeToString(hashsum)
+	}
+	for fileName, hashsum := range stateDBExportSummary {
+		filesAndHashes[fileName] = hex.EncodeToString(hashsum)
+	}
+	bcInfo, err := l.GetBlockchainInfo()
+	if err != nil {
+		return err
+	}
+	metadata, err := json.MarshalIndent(
+		&snapshotSignableMetadata{
+			ChannelName:        l.ledgerID,
+			ChannelHeight:      bcInfo.Height,
+			LastBlockHashInHex: hex.EncodeToString(bcInfo.CurrentBlockHash),
+			FilesAndHashes:     filesAndHashes,
+		},
+		"",
+		jsonFileIndent,
+	)
+	if err != nil {
+		return errors.Wrap(err, "error while marshelling snapshot metadata to JSON")
+	}
+	if err := createAndSyncFile(filepath.Join(dir, snapshotMetadataFileName), metadata); err != nil {
+		return err
+	}
+
+	// generate metadata hash file
+	hash, err := l.hashProvider.GetHash(snapshotHashOpts)
+	if err != nil {
+		return err
+	}
+	if _, err := hash.Write(metadata); err != nil {
+		return err
+	}
+	metadataAdditionalInfo, err := json.MarshalIndent(
+		&snapshotAdditionalInfo{
+			SnapshotHashInHex:        hex.EncodeToString(hash.Sum(nil)),
+			LastBlockCommitHashInHex: hex.EncodeToString(l.commitHash),
+		},
+		"",
+		jsonFileIndent,
+	)
+	if err != nil {
+		return errors.Wrap(err, "error while marshalling snapshot additional info to JSON")
+	}
+	if err := createAndSyncFile(filepath.Join(dir, snapshotMetadataHashFileName), metadataAdditionalInfo); err != nil {
+		return err
+	}
+	return nil
+}
+
+func createAndSyncFile(filePath string, content []byte) error {
+	file, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0444)
+	if err != nil {
+		return errors.Wrapf(err, "error while creating file:%s", filePath)
+	}
+	_, err = file.Write(content)
+	if err != nil {
+		file.Close()
+		return errors.Wrapf(err, "error while writing to file:%s", filePath)
+	}
+	if err = file.Sync(); err != nil {
+		file.Close()
+		return errors.Wrapf(err, "error while synching the file:%s", filePath)
+	}
+	if err := file.Close(); err != nil {
+		return errors.Wrapf(err, "error while closing the file:%s", filePath)
+	}
+	return nil
+}
+
+func syncParentDir(path string) error {
+	return syncDir(filepath.Dir(path))
+}
+
+func syncDir(dirPath string) error {
+	dir, err := os.Open(dirPath)
+	if err != nil {
+		return errors.Wrapf(err, "error while opening dir:%s", dirPath)
+	}
+	err = dir.Sync()
+	if err != nil {
```

Comment:
Nit. 

At a few places, there is some inconsistent usage of 
```
err = XXX()
if err != nil {
```
```
if err = XXX(); err != nil {
```
At least for the smaller length lines, we can use one type consistently.